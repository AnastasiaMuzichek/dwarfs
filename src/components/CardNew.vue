<script setup lang="ts">
import { Card, CardMeta, Button } from 'ant-design-vue';
import { PlusOutlined } from '@ant-design/icons-vue';
import type { PropType } from 'vue';
import type { DwarfValue } from '../models/dwarf';

const emit = defineEmits<{
    (event: 'edit', dwarf: DwarfValue): void;
}>();
defineProps({
    dwarf: {
        required: true,
        type: Object as PropType<DwarfValue>,
    },
});
</script>

<template>
    <Card hoverable style="width: 250px; height: 445px">
        <template #cover>
            <Button @click="emit('edit', dwarf)" type="link">
                <template #icon>
                    <PlusOutlined
                        style="color: rgb(3, 168, 168)"
                    ></PlusOutlined>
                </template>
            </Button>
        </template>
        <CardMeta title="Добавить гнома"> </CardMeta>
    </Card>
</template>

<style scoped lang="less">
:deep(.ant-card-cover .ant-btn) {
    height: 300px;
    width: 100%;
    font-size: 80px;
}
</style>
