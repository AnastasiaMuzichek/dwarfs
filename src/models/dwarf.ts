import type { WeaponValue } from './weapon';

export type DwarfValue = {
    id: number;
    image: string;
    name: string;
    age: number;
    skill: string;
    history: string;
    weapon: WeaponValue;
};

export function createDwarf(data: Record<string, any>): DwarfValue {
    return {
        id: data.id,
        image: data.image,
        name: data.name,
        age: data.age || 0,
        skill: data.skill,
        history: data.history,
        weapon: data.weapon,
    };
}
