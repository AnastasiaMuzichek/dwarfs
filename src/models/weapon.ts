export type WeaponValue = {
    description: string;
    image: string;
    name: string;
    value: string;
};

export function createWeapon(data: Record<string, any>): WeaponValue {
    return {
        description: data.description,
        image: data.image,
        name: data.name,
        value: data.value,
    };
}
