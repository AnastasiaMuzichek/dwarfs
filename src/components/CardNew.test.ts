import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import CardNew from './CardNew.vue';
import { createDwarf } from '../models/dwarf';

describe('CardNew render', () => {
    it('Default render', () => {
        const dwarf = createDwarf({ id: null });
        const wrapper = mount(CardNew, {
            props: { dwarf },
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});
