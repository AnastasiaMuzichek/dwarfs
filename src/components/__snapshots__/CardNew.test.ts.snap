// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardNew render > Default render 1`] = `
<div
  class="ant-card css-dev-only-do-not-override-1p3hq3p ant-card-bordered ant-card-hoverable"
  data-v-42e0d835=""
  style="width: 250px; height: 445px;"
>
  <!---->
  <div
    class="ant-card-cover"
  >
    
    <button
      class="css-dev-only-do-not-override-1p3hq3p ant-btn ant-btn-link ant-btn-icon-only"
      data-v-42e0d835=""
      type="button"
    >
      
      <span
        aria-label="plus"
        class="anticon anticon-plus"
        data-v-42e0d835=""
        role="img"
        style="color: rgb(3, 168, 168);"
      >
        <svg
          aria-hidden="true"
          data-icon="plus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
          />
          <path
            d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
          />
        </svg>
        <!---->
      </span>
      
      
      
    </button>
    
  </div>
  <div
    class="ant-card-body"
  >
    
    <div
      class="ant-card-meta"
      data-v-42e0d835=""
    >
      <!---->
      <div
        class="ant-card-meta-detail"
      >
        <div
          class="ant-card-meta-title"
        >
          Добавить гнома
        </div>
        <!---->
      </div>
    </div>
    
  </div>
  <!---->
</div>
`;
